share/normalize/links.py

```diff
@@ -368,7 +409,37 @@ def execute(self, obj):
 
 class TextLink(AbstractLink):
     def execute(self, obj):
-        return obj.text
+        if isinstance(obj, str):
+            return obj
+        elif isinstance(obj, dict):
+            if '#text' in obj:
+                return obj['#text']
+            raise Exception('#text is not in {}'.format(obj))
+        elif isinstance(obj, list):
+            for item in obj:
+                if isinstance(item, str):
+                    return item
+            raise logger.warning('No value in list {} is a string.'.format(obj))
```

Comment:
Unfortunately, raising an exception here will cause some issues because if the values are highly nested then we don't want them. For instance, titles can be abstracts, alternate, or translated but the main title is usually (but not always) just a string in the list.